src/codegen.cpp

```diff
@@ -434,6 +434,17 @@ static bool isbits_spec(jl_value_t *jt, bool allow_singleton = true)
         (allow_singleton || (jl_datatype_size(jt) > 0) || (jl_datatype_nfields(jt) > 0));
 }
 
+static MDNode *best_tbaa(jl_value_t *jt) {
+    jt = jl_unwrap_unionall(jt);
+    if (!jl_is_datatype(jt))
+        return tbaa_value;
+    if (jl_is_abstracttype(jt))
+        return tbaa_value;
```

Comment:
Since these conditions return the same thing, couldn't you just collapse them as
```c
if (!jl_is_datatype(jt) || jl_is_abstracttype(jt))
    return tbaa_value;
```
?